packages/router/src/router.js

```diff
@@ -16,6 +16,24 @@ const Route = () => {
   return null
 }
 
+const PrivateRoute = (props) => {
+  return <Route private {...props} />
+}
+
+const PrivatePageLoader = ({ useAuth, children }) => {
+  const { loading, authenticated } = useAuth()
+
+  if (loading) {
+    return null
+  }
+
+  if (authenticated) {
+    return children
+  } else {
+    // TODO: Should we navigate away, exec a function?
```

Comment:
Exec a function sounds good. Thinking out loud, maybe pages could export an `unauthorized` render func? Similar to how Cells work?

Reply:
That would be cool, and is a nice way of dealing with a single page, but I was thinking of adding a prop to the Router since it is probably something that you want to do for all private routes, rather than just the one.

I absolutely think we should have an option for a single route, but we would also need an option for all the routes.

Reply:
Good point on nested routes, top level prop fn makes sense. 

I wonder if we could have a default functionality that also shows a `NotAuthorized` page, similar to the existing `NotFound` and `FatalError` pages.

Reply:
Ooooh! I like that!